packages/wallet/backend/src/socket/service.ts

```diff
@@ -0,0 +1,60 @@
+import { env, type Env } from '@/config/env'
+import { Logger } from 'winston'
+import socketIo from 'socket.io'
+import http from 'http'
+import { unsealData } from 'iron-session'
+import { AccountService } from '@/account/service'
+
+interface ISocketService {
+  init(args: http.Server): void
+  emitAccountsUpdateById(id: string): void
+}
+
+interface SocketServiceDependencies {
+  logger: Logger
+  env: Env
+  accountService: AccountService
+}
+
+export class SocketService implements ISocketService {
+  private io: socketIo.Server | null = null
+
+  constructor(private deps: SocketServiceDependencies) {}
+
+  init(server: http.Server) {
+    this.io = new socketIo.Server(server, {
+      cors: {
+        origin: [
+          'http://localhost:4003',
+          `https://${this.deps.env.RAFIKI_MONEY_FRONTEND_HOST}`
+        ],
+        credentials: true
+      }
+    })
+    this.deps.logger.info(`Socket Server is started...`)
+
+    this.io.on('connection', async (socket) => {
+      const token = socket.handshake.query.token?.toString().split('=')[1]
```

Comment:
can the token be sent in headers or cookies?

Reply:
hmm, in my understanding handshake is used for this purpose in socket communication. how do you think @beniaminmunteanu ?

Reply:
I also think it should be in the auth headers, and checked on the handshake handle. 
 The client already has the auth token stored when the user logs in. the same token should be used when creating the websocket connection, and as such checked at handshake time.